<launch>
    <arg name="gui" default="False" />  

    <include file="$(find val_logic_manager)/launch/UnitA_robot_description.launch"/>

    <group ns="val_ik_finalpose_robot">
      <param name="robot_description" textfile="$(find val_viz)/robot_urdf/valkyrie_sim_gazebo_sync_red.urdf" />        
      <param name="use_gui" value="$(arg gui)"/>

      <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
          <rosparam param="source_list">[/val_ik_finalpose_robot/joint_states]</rosparam> 
      </node>

      <node pkg="robot_state_publisher" type="state_publisher" name="robot_state_publisher">
        <param name="tf_prefix" type="string" value="val_ik_finalpose_robot" />  
      </node>
    </group>

    <group ns="val_ik_initpose_robot">
      <param name="robot_description" textfile="$(find val_viz)/robot_urdf/valkyrie_sim_gazebo_sync_red.urdf" />        
      <param name="use_gui" value="$(arg gui)"/>

      <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
          <rosparam param="source_list">[/val_ik_initpose_robot/joint_states]</rosparam> 
      </node>

      <node pkg="robot_state_publisher" type="state_publisher" name="robot_state_publisher">
        <param name="tf_prefix" type="string" value="val_ik_initpose_robot" />  
      </node>
    </group>


    <node name="rviz_ik" pkg="rviz" type="rviz" args="-d $(find val_logic_manager)/launch/logic_manager.rviz" required="true" /> 

</launch>